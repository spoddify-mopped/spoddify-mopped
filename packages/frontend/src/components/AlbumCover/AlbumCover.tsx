import { Album } from '../../clients/api.types';
import Cover from '../Cover/Cover';
import React from 'react';

type Props = {
  item: Album;
  onClick: (id: string) => void;
};

const AlbumCover = ({ item, onClick }: Props): React.ReactElement => {
  return (
    <Cover
      item={{
        id: item.id,
        image: item.imageUrl,
        name: item.name,
        onClick: () => {
          onClick(item.id);
        },
        subtitle: <span>{new Date(item.releaseDate || '').getFullYear()}</span>,
      }}
    />
  );
};

export default AlbumCover;
