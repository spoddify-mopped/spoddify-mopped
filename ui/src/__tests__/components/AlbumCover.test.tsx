import AlbumCover from '../../components/AlbumCover/AlbumCover';
import React from 'react';
import renderer from 'react-test-renderer';

it('AlbumCover renders correctly with release date', () => {
  const tree = renderer
    .create(
      <AlbumCover
        item={{
          artists: [
            {
              id: 'some artist id',
              name: 'some artist name',
            },
          ],
          id: 'some id',
          name: 'some name',
          releaseDate: '2014-01-01',
        }}
        onClick={() => ''}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('AlbumCover renders correctly with release date and image', () => {
  const tree = renderer
    .create(
      <AlbumCover
        item={{
          artists: [
            {
              id: 'some artist id',
              name: 'some artist name',
            },
          ],
          id: 'some id',
          imageUrl: 'some image url',
          name: 'some name',
          releaseDate: '2014-01-01',
        }}
        onClick={() => ''}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
