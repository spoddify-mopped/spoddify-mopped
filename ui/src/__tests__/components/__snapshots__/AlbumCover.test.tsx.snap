// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumCover renders correctly with release date 1`] = `
<div
  className="cover "
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="cover_replacement.png"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  >
    <span>
      2014
    </span>
  </span>
</div>
`;

exports[`AlbumCover renders correctly with release date and image 1`] = `
<div
  className="cover "
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="some image url"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  >
    <span>
      2014
    </span>
  </span>
</div>
`;
